### ISSUE TITLE
`MemberNotNullWhen` does not report "Member must have a non-null value when exiting when" when expression is returned 

### ISSUE THREAD

**Issue by gurustron** [dotnet/roslyn#70398]
**Version Used**: 
7.0.400
**Steps to Reproduce**:

```c#
public class Test
{
	public string? Foo { get; set; }
	public string? Bar { get; set; }

	[MemberNotNullWhen(true, nameof(Foo), nameof(Bar))]
	public bool IsInitialized => this.Foo != null;
}
public class Test1
{
	public string? Foo { get; set; }
	public string? Bar { get; set; }

	[MemberNotNullWhen(true, nameof(Foo), nameof(Bar))]
	public bool IsInitialized
	{
		get { return this.Foo != null; }
	}
}
public class Test2
{
	public string? Foo { get; set; }
	public string? Bar { get; set; }

	[MemberNotNullWhen(true, nameof(Foo), nameof(Bar))]
	public bool IsInitialized
	{
		get
		{
			if (this.Foo != null)
			{
				return true;
			}

			return false;
		}
	}
}
```
[sharplab](https://sharplab.io/#v2:D4AQTAjAsAUCAMACEEB0ARAlgQwOYDsB7AZwBdMBjY1AYUIBMBTAQX2wBsBPYzYgbliwUATgAUAIkA8G4BB98QEoBMQXADMyMIgAqjMrADesAJAg1KeAH5EAMUKFEexLkak+iYs9cBfIyeQQLiABC2ABO9o4ebpHeSjCGANoAsowAtgBGjCEAcoSkWQCu7OwA6gAWjPiipCH5jAA0iGwpjIQAZqI2hHINTS3twSFycgC6PmpptuyIAJLE0/iY5ByYAF6M9IgAvAB8iKSlvKidiACEm42F7IoxvuBaOqTQMAZxt/6Wxw5OLlE/Mcamd5BULhb6udx/ZQJZLpTI5PKXMoVKo1eqNbDNNodWzddGY/qhIajV7jSYzOYLJbsVbrIwvQyGb7hEAAdj2B2oxzOFyKXiMMRupg02jIYH0Yz8AU+ER+EL5JMllgGoMicsQMSMSVSGWyuQKRSRlWqtR6GL62K6pvxogGRIlE0IU1m80WOGpa3odKMjOc3vpDMwrUQVQ5R1sp3O+Eucm9hn9DNZe1RigZhg1cQTbNaHHcKbT/NgniAA===)

**Expected Behavior**:

CS8774/CS8775 "Member must have a non-null value when exiting." should be reported in all 3 cases

**Actual Behavior**:

Only explicit `if` expansion results in CS8775
